mesa/space.py

```diff
@@ -256,29 +254,37 @@ def get_neighborhood(
         if neighborhood is not None:
             return neighborhood
 
-        coordinates: set[Coordinate] = set()
-
         x, y = pos
-        for dy in range(-radius, radius + 1):
-            for dx in range(-radius, radius + 1):
-                # Skip coordinates that are outside manhattan distance
-                if not moore and abs(dx) + abs(dy) > radius:
-                    continue
 
-                coord = (x + dx, y + dy)
+        x_ranges = self._compute_inbound_ranges(x, self.width, radius)
+        y_ranges = self._compute_inbound_ranges(y, self.height, radius)
 
-                if self.out_of_bounds(coord):
-                    # Skip if not a torus and new coords out of bounds.
-                    if not self.torus:
-                        continue
-                    coord = self.torus_adj(coord)
+        neighborhood = []
 
-                coordinates.add(coord)
+        for x_range, y_range in itertools.product(x_ranges, y_ranges):
```

Comment:
Why do we now require 3 levels of loop, where previously it was just 2?

Reply:
because there is the possibility that there two different ranges for x and/or y in the torus case, so you need to make all the combinations of ranges

Reply:
for example with a grid 10x10 radius 1 from (0,0) gives you range(0,1) and range(9,10)

Reply:
or we could chain together the ranges to solve this

Reply:
I realized there is no way to change this because itertools.chain generates an iterator which is exausted after the first iteration but it can happen to iterate more than one time

Reply:
There might be a stdlib function that can concatenate the range objects.

Reply:
Searched it and didn't find anything unfortunately